### PR TITLE
✨ feat: add Gitee AI model provider

### PR DESCRIPTION
**Pull request by hezhijie0327**
#### 💻 变更类型 | Change Type

- [X] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] 📝 docs
- [ ] 🔨 chore

#### 🔀 变更说明 | Description of Change
1. 模型供应商图标添加 https://github.com/lobehub/lobe-icons/issues/53
2. 禁用客户端请求模式（不支持）
3. 引入自定义变量 `GITEE_AI_API_KEY` `GITEE_AI_MODEL_LIST`

Note:
1. `code-raccoon-v1` 对话流存在结构问题
```
{"data":{"id":"9d58d703-7f48-4c21-b09e-001b9fb47d81","usage":{"prompt_tokens":1,"completion_tokens":10,"knowledge_tokens":0,"total_tokens":11},"choices":[{"index":0,"role":"assistant","message":"Hello! How can I assist you today?\u003c|endofblock|\u003e","finish_reason":"stop","text":""}],"plugins":{}}}
```
2. 获取 API Key https://ai.gitee.com
image

#### 📝 补充信息 | Additional Information
|模块|截图|
|-|-|
|模型列表|imageimage|
|对话|image|
|工具调用|image|

### CODE REVIEW

**Comment by bluishoul** [comment]
Hi @hezhijie0327, thanks for adding support for Gitee AI, I'm working on the Gitee AI project, and I have some change request for the PR for some branding stuff. Could you please change `Gitee` to `Gitee AI`, for example: 

- Variables: 
```diff
- GITEE_API_KEY="" GITEE_MODEL_LIST=""
+ GITEE_AI_API_KEY="" GITEE_AI_MODEL_LIST=""
```

- Filename:
```diff
- src/config/modelProviders/gitee.ts
+ src/config/modelProviders/giteeai.ts
```

I will send my review as far as I could (there no good way to change the filename in the PR directly).

Thank you.

File: Dockerfile [hunk 1]
```diff
@@ -158,6 +158,8 @@ ENV \
     DEEPSEEK_API_KEY="" DEEPSEEK_MODEL_LIST="" \
     # Fireworks AI
     FIREWORKSAI_API_KEY="" FIREWORKSAI_MODEL_LIST="" \
+    # Gitee
+    GITEE_API_KEY="" GITEE_MODEL_LIST="" \
```
Comment (bluishoul) [inline Dockerfile]:
```suggestion
    # Gitee AI
    GITEE_AI_API_KEY="" GITEE_AI_MODEL_LIST="" \
```

File: Dockerfile.database [hunk 2]
```diff
@@ -193,6 +193,8 @@ ENV \
     DEEPSEEK_API_KEY="" DEEPSEEK_MODEL_LIST="" \
     # Fireworks AI
     FIREWORKSAI_API_KEY="" FIREWORKSAI_MODEL_LIST="" \
+    # Gitee
+    GITEE_API_KEY="" GITEE_MODEL_LIST="" \
```
Comment (bluishoul) [inline Dockerfile.database]:
```suggestion
    # Gitee AI
    GITEE_AI_API_KEY="" GITEE_AI_MODEL_LIST="" \
```

File: src/app/(main)/settings/llm/ProviderList/providers.tsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ import {
   BaichuanProviderCard,
   DeepSeekProviderCard,
   FireworksAIProviderCard,
+  GiteeProviderCard,
```
Comment (bluishoul) [inline src/app/(main)/settings/llm/ProviderList/providers.tsx]:
```suggestion
  GiteeAIProviderCard,
```

File: src/app/(main)/settings/llm/ProviderList/providers.tsx [hunk 4]
```diff
@@ -88,6 +89,7 @@ export const useProviderList = (): ProviderItem[] => {
       TaichuProviderCard,
       InternLMProviderCard,
       SiliconCloudProviderCard,
+      GiteeProviderCard,
```
Comment (bluishoul) [inline src/app/(main)/settings/llm/ProviderList/providers.tsx]:
```suggestion
      GiteeAIProviderCard,
```

File: src/config/llm.ts [hunk 5]
```diff
@@ -102,6 +102,9 @@ export const getLLMConfig = () => {
       SILICONCLOUD_API_KEY: z.string().optional(),
       SILICONCLOUD_PROXY_URL: z.string().optional(),
 
+      ENABLED_GITEE: z.boolean(),
+      GITEE_API_KEY: z.string().optional(),
```
Comment (bluishoul) [inline src/config/llm.ts]:
```suggestion
      ENABLED_GITEE_AI: z.boolean(),
      GITEE_AI_API_KEY: z.string().optional(),
```

File: src/config/llm.ts [hunk 6]
```diff
@@ -227,6 +230,9 @@ export const getLLMConfig = () => {
       SILICONCLOUD_API_KEY: process.env.SILICONCLOUD_API_KEY,
       SILICONCLOUD_PROXY_URL: process.env.SILICONCLOUD_PROXY_URL,
 
+      ENABLED_GITEE: !!process.env.GITEE_API_KEY,
+      GITEE_API_KEY: process.env.GITEE_API_KEY,
```
Comment (bluishoul) [inline src/config/llm.ts]:
```suggestion
      ENABLED_GITEE_AI: !!process.env.GITEE_AI_API_KEY,
      GITEE_AI_API_KEY: process.env.GITEE_AI_API_KEY,
```

File: src/config/modelProviders/gitee.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
+import { ModelProviderCard } from '@/types/llm';
+
+// ref: https://ai.gitee.com/serverless-api/packages/1493
+const Gitee: ModelProviderCard = {
```
Comment (bluishoul) [inline src/config/modelProviders/gitee.ts]:
```suggestion
const GiteeAI: ModelProviderCard = {
```

**Comment by hezhijie0327** [comment]
Sure. Will update later

File: src/config/modelProviders/gitee.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      enabled: true,
+      id: 'codegeex4-all-9b',
+      tokens: 40_000,
+    },
+  ],
+  checkModel: 'Qwen2-7B-Instruct',
+  description:
+    'Gitee AI 的 Serverless API 为 AI 开发者提供开箱即用的大模型推理 API 服务。',
+  disableBrowserRequest: true,
+  id: 'gitee',
```
Comment (bluishoul) [inline src/config/modelProviders/gitee.ts]:
```suggestion
  id: 'giteeai',
```

File: src/config/modelProviders/gitee.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  modelList: { showModelFetcher: true },
+  modelsUrl: 'https://ai.gitee.com/docs/openapi/v1#tag/serverless/POST/chat/completions',
+  name: 'Gitee AI',
+  url: 'https://ai.gitee.com',
+};
+
+export default Gitee;
```
Comment (bluishoul) [inline src/config/modelProviders/gitee.ts]:
```suggestion
export default GiteeAI;
```

File: src/config/modelProviders/index.ts [hunk 10]
```diff
@@ -9,6 +9,7 @@ import BedrockProvider from './bedrock';
 import CloudflareProvider from './cloudflare';
 import DeepSeekProvider from './deepseek';
 import FireworksAIProvider from './fireworksai';
+import GiteeProvider from './gitee';
```
Comment (bluishoul) [inline src/config/modelProviders/index.ts]:
```suggestion
import GiteeAIProvider from './giteeai';
```

File: src/config/modelProviders/index.ts [hunk 11]
```diff
@@ -64,6 +65,7 @@ export const LOBE_DEFAULT_MODEL_LIST: ChatModelCard[] = [
   CloudflareProvider.chatModels,
   Ai360Provider.chatModels,
   SiliconCloudProvider.chatModels,
+  GiteeProvider.chatModels,
```
Comment (bluishoul) [inline src/config/modelProviders/index.ts]:
```suggestion
  GiteeAIProvider.chatModels,
```

File: src/config/modelProviders/index.ts [hunk 12]
```diff
@@ -109,6 +111,7 @@ export const DEFAULT_MODEL_PROVIDER_LIST = [
   TaichuProvider,
   InternLMProvider,
   SiliconCloudProvider,
+  GiteeProvider,
```
Comment (bluishoul) [inline src/config/modelProviders/index.ts]:
```suggestion
  GiteeAIProvider,
```

File: src/config/modelProviders/index.ts [hunk 13]
```diff
@@ -129,6 +132,7 @@ export { default as BedrockProviderCard } from './bedrock';
 export { default as CloudflareProviderCard } from './cloudflare';
 export { default as DeepSeekProviderCard } from './deepseek';
 export { default as FireworksAIProviderCard } from './fireworksai';
+export { default as GiteeProviderCard } from './gitee';
```
Comment (bluishoul) [inline src/config/modelProviders/index.ts]:
```suggestion
export { default as GiteeAIProviderCard } from './giteeai';
```

File: src/libs/agent-runtime/AgentRuntime.ts [hunk 14]
```diff
@@ -12,6 +12,7 @@ import { LobeBedrockAI, LobeBedrockAIParams } from './bedrock';
 import { LobeCloudflareAI, LobeCloudflareParams } from './cloudflare';
 import { LobeDeepSeekAI } from './deepseek';
 import { LobeFireworksAI } from './fireworksai';
+import { LobeGiteeAI } from './gitee';
```
Comment (bluishoul) [inline src/libs/agent-runtime/AgentRuntime.ts]:
```suggestion
import { LobeGiteeAI } from './giteeai';
```

File: src/libs/agent-runtime/AgentRuntime.ts [hunk 15]
```diff
@@ -137,6 +138,7 @@ class AgentRuntime {
       cloudflare: Partial<LobeCloudflareParams>;
       deepseek: Partial<ClientOptions>;
       fireworksai: Partial<ClientOptions>;
+      gitee: Partial<ClientOptions>;
```
Comment (bluishoul) [inline src/libs/agent-runtime/AgentRuntime.ts]:
```suggestion
      giteeai: Partial<ClientOptions>;
```

File: src/libs/agent-runtime/AgentRuntime.ts [hunk 16]
```diff
@@ -303,6 +305,11 @@ class AgentRuntime {
         break;
       }
 
+      case ModelProvider.Gitee: {
+        runtimeModel = new LobeGiteeAI(params.gitee);
```
Comment (bluishoul) [inline src/libs/agent-runtime/AgentRuntime.ts]:
```suggestion
        runtimeModel = new LobeGiteeAI(params.giteeai);
```

File: src/libs/agent-runtime/gitee/index.test.ts [hunk 17]
```diff
@@ -0,0 +1,255 @@
...
+import {
+  ChatStreamCallbacks,
+  LobeOpenAICompatibleRuntime,
+  ModelProvider,
+} from '@/libs/agent-runtime';
+
+import * as debugStreamModule from '../utils/debugStream';
+import { LobeGiteeAI } from './index';
+
+const provider = ModelProvider.Gitee;
```
Comment (bluishoul) [inline src/libs/agent-runtime/gitee/index.test.ts]:
```suggestion
const provider = ModelProvider.GiteeAI;
```

File: src/libs/agent-runtime/gitee/index.test.ts [hunk 18]
```diff
@@ -0,0 +1,255 @@
...
+          stream: true,
+          temperature: 0,
+        });
+
+        // 验证 debugStream 被调用
+        expect(debugStreamModule.debugStream).toHaveBeenCalled();
+
+        // 恢复原始环境变量值
+        process.env.DEBUG_GITEE_CHAT_COMPLETION = originalDebugValue;
+      });
```
Comment (bluishoul) [inline src/libs/agent-runtime/gitee/index.test.ts]:
```suggestion
      it('should call debugStream and return StreamingTextResponse when DEBUG_GITEE_AI_CHAT_COMPLETION is 1', async () => {
        // Arrange
        const mockProdStream = new ReadableStream() as any; // 模拟的 prod 流
        const mockDebugStream = new ReadableStream({
          start(controller) {
            controller.enqueue('Debug stream content');
            controller.close();
          },
        }) as any;
        mockDebugStream.toReadableStream = () => mockDebugStream; // 添加 toReadableStream 方法

        // 模拟 chat.completions.create 返回值，包括模拟的 tee 方法
        (instance['client'].chat.completions.create as Mock).mockResolvedValue({
          tee: () => [mockProdStream, { toReadableStream: () => mockDebugStream }],
        });

        // 保存原始环境变量值
        const originalDebugValue = process.env.DEBUG_GITEE_AI_CHAT_COMPLETION;

        // 模拟环境变量
        process.env.DEBUG_GITEE_AI_CHAT_COMPLETION = '1';
        vi.spyOn(debugStreamModule, 'debugStream').mockImplementation(() => Promise.resolve());

        // 执行测试
        // 运行你的测试函数，确保它会在条件满足时调用 debugStream
        // 假设的测试函数调用，你可能需要根据实际情况调整
        await instance.chat({
          messages: [{ content: 'Hello', role: 'user' }],
          model: 'Qwen2-7B-Instruct',
          stream: true,
          temperature: 0,
        });

        // 验证 debugStream 被调用
        expect(debugStreamModule.debugStream).toHaveBeenCalled();

        // 恢复原始环境变量值
        process.env.DEBUG_GITEE_AI_CHAT_COMPLETION = originalDebugValue;
      });
```

File: src/libs/agent-runtime/gitee/index.ts [hunk 19]
```diff
@@ -0,0 +1,10 @@
+import { ModelProvider } from '../types';
+import { LobeOpenAICompatibleFactory } from '../utils/openaiCompatibleFactory';
+
+export const LobeGiteeAI = LobeOpenAICompatibleFactory({
+  baseURL: 'https://ai.gitee.com/v1',
+  debug: {
+    chatCompletion: () => process.env.DEBUG_GITEE_CHAT_COMPLETION === '1',
```
Comment (bluishoul) [inline src/libs/agent-runtime/gitee/index.ts]:
```suggestion
    chatCompletion: () => process.env.DEBUG_GITEE_AI_CHAT_COMPLETION === '1',
```

File: src/libs/agent-runtime/types/type.ts [hunk 20]
```diff
@@ -31,6 +31,7 @@ export enum ModelProvider {
   Cloudflare = 'cloudflare',
   DeepSeek = 'deepseek',
   FireworksAI = 'fireworksai',
+  Gitee = 'gitee',
```
Comment (bluishoul) [inline src/libs/agent-runtime/types/type.ts]:
```suggestion
  GiteeAI = 'giteeai',
```

File: src/server/modules/AgentRuntime/index.ts [hunk 21]
```diff
@@ -236,6 +236,13 @@ const getLlmOptionsFromPayload = (provider: string, payload: JWTPayload) => {
 
       return { apiKey, baseURL };
     }
+    case ModelProvider.Gitee: {
+      const { GITEE_API_KEY } = getLLMConfig();
+
+      const apiKey = apiKeyManager.pick(payload?.apiKey || GITEE_API_KEY);
+
+      return { apiKey };
+    }
```
Comment (bluishoul) [inline src/server/modules/AgentRuntime/index.ts]:
```suggestion
    case ModelProvider.GiteeAI: {
      const { GITEE_AI_API_KEY } = getLLMConfig();

      const apiKey = apiKeyManager.pick(payload?.apiKey || GITEE_AI_API_KEY);

      return { apiKey };
    }
```

File: src/types/user/settings/keyVaults.ts [hunk 22]
```diff
@@ -41,6 +41,7 @@ export interface UserKeyVaults {
   cloudflare?: CloudflareKeyVault;
   deepseek?: OpenAICompatibleKeyVault;
   fireworksai?: OpenAICompatibleKeyVault;
+  gitee?: OpenAICompatibleKeyVault;
```
Comment (bluishoul) [inline src/types/user/settings/keyVaults.ts]:
```suggestion
  giteeai?: OpenAICompatibleKeyVault;
```

Comment (hezhijie0327) [reply src/types/user/settings/keyVaults.ts]: Done

File: src/libs/agent-runtime/togetherai/__snapshots__/index.test.ts.snap
Comment (arvinxx) [inline src/libs/agent-runtime/togetherai/__snapshots__/index.test.ts.snap:1]: 奇怪…为啥这边的snapshot变了

Comment (hezhijie0327) [reply src/libs/agent-runtime/togetherai/__snapshots__/index.test.ts.snap:1]: 模型名字重了，然后 CI 炸了...
```
    {
      description: 'Yi-1.5-34B 在保持原系列模型优秀的通用语言能力的前提下，通过增量训练 5 千亿高质量 token，大幅提高了数学逻辑、代码能力。',
      displayName: 'Yi 34B Chat',
      enabled: true,
      id: 'Yi-34B-Chat',
      tokens: 4000,
    },
```

**Pull request merged**